### DISCUSSION TITLE
Meaning of different execute_on parameters

### DISCUSSION THREAD

**Discussion by js-jixu** [idaholab/moose#22667] [Q&A General] [answered]
Hi, experts.

Many objects have the parameter `execute_on`, and `execute_on` also has many options. Some options have obvious meanings, such as `TIMESTEP_END`. But I also have doubts about some options.

1. I guess `LINEAR` and `NONLINEAR` stand for executing the object on linear and non-linear iterations respectively. Is my guess right?

2. Does `ALWAYS` mean that object is always executed in the steady or transient simulation? That sounds unlikely.

3. How is `CUSTOM` used? It looks like `CUSTOM` needs some other parameters to help.

**Comment by hugary1995** [comment, marked as answer]
Detailed information here: https://mooseframework.inl.gov/moose/source/interfaces/SetupInterface.html

The only ones not so intuitive are LINEAR and NONLINEAR, they actually mean RESIDUAL and JACOBIAN, resp.

**Reply by GiudGiud** [reply]
CUSTOM is on its way out because apps can now register additional execute_on flags
https://github.com/idaholab/moose/issues/22124

**Reply by js-jixu** [reply]
Thank you for replies! I'm reading those.

**Reply by maxnezdyur** [reply]
Just to confirm. When using multiapp and TIMESTEP_BEGIN, an object will be called every fixed point iteration within a timestep if they have a TIMESTEP_BEGIN flag active?
